### ISSUE TITLE
[BUG]Set initialTopMostItemIndex with last index, but first item renderred in first frame and exists scroll

### ISSUE THREAD

**Issue by stefanJi** [petyosi/react-virtuoso#888]
I don't provide free support through Github issues. If you want to ask for support or request features, [sponsor the project](https://github.com/sponsors/petyosi) and contact me over email.
If you choose to ignore the above, I am likely going to delete the issue. 

**Screenshots**

https://user-images.githubusercontent.com/15273198/234493991-b1c34c88-35bf-4456-a077-1773257df1dc.mov

**Desktop (please complete the following information):**
 - OS: MacOS.
 - Browser Electron(chrome)

**Additional context**

**Comment by stefanJi** [comment]
version: 4.1.0
```
 <Virtuoso
            firstItemIndex={firstItemIndex}
            initialTopMostItemIndex={messages.length - 1}
            data={messages}
            itemContent={itemContent}
            followOutput={'smooth'}
        />
```

**Comment by petyosi** [comment]
@stefanJi - I am sorry, I cannot comprehend what your problem is.

**Comment by petyosi** [comment]
I can't debug videos. Follow the issue template and create a reproduction in codesandbox. Make it as minimal as possible.

**Comment by stefanJi** [comment]
@petyosi This is a reproduction in codesandbox https://codesandbox.io/s/virtuoso-demo-qw9zf7?file=/src/App.tsx

---

When switching sessions, you can see a quick flick phenomenon. By dividing the screencast video into frames, it can be found that the content of index 0 is rendered first before the final content when switching sessions.

![20230509-104014](https://user-images.githubusercontent.com/15273198/236980181-a4b66669-2daf-4a7b-b588-7e2272a7eac3.jpeg)

![20230509-104020](https://user-images.githubusercontent.com/15273198/236980203-a286999b-8bc8-460e-907c-76cf0ff8e018.jpeg)

---

This is screen record

https://user-images.githubusercontent.com/15273198/236980074-af7adf9a-b493-477a-bc87-c3d8f9e850fd.mov

**Comment by petyosi** [comment]
I believe this got solved in v4.3.6.
